### PR TITLE
Use gotestsum to publish test results in JUnit format

### PR DESCRIPTION
**Pull request by mem**
Signed-off-by: Marcelo E. Magallon <marcelo@sylabs.io>

## Description of the Pull Request (PR):

Write your description of the PR here. Be sure to include as much background,
and details necessary for the reviewers to understand exactly what this is
fixing or enhancing.

### This fixes or addresses the following GitHub issues:

 - Fixes #

#### Before submitting a PR, make sure you have done the following:

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/master/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

**Review by dtrudg — APPROVED** [review]
LGTM.

I got some nice unit-test.xml.json output when running with `JUNIT_OUTPUT` with the slight delay while my slow brain realized I had to have gotestsum in the sudo `PATH` not just my `PATH`

File: scripts/go-test.in [hunk 1]
```diff
@@ -21,17 +28,50 @@ for arg in "$@" ; do
 			skip=true
 			;;
 
+		-junit)
+			if ! command -v gotestsum 2> /dev/null ; then
+				echo 'E: JUnit output requested but gotestsum not found in PATH. Abort.'
```
Comment (dtrudg) [inline scripts/go-test.in]: I was stupidly looking on my PATH for a bit, not the PATH sudo has... could clarify here? But that's a tiny minor nitpick as this is a CI focused change.

Comment (mem) [reply scripts/go-test.in]: Would printing out the PATH work?

Comment (dtrudg) [reply scripts/go-test.in]: I think that should make it pretty obvious, yes.

**Pull request merged**